src/Form/QueueContentsForm.php

```diff
@@ -0,0 +1,173 @@
+<?php
+
+namespace Drupal\media_mpx\Form;
+
+use Drupal\Core\Batch\BatchBuilder;
+use Drupal\Core\Form\FormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\media_mpx\Repository\MpxMediaType;
+use Drupal\media_mpx\Service\QueueVideoImportsRequest;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Class QueueContentsForm.
+ */
+class QueueContentsForm extends FormBase {
+
+  /**
+   * The mpx Media Types repository.
+   *
+   * @var \Drupal\media_mpx\Repository\MpxMediaType
+   */
+  private $mpxMediaTypeRepository;
+
+  /**
+   * QueueContentsForm constructor.
+   */
+  public function __construct(MpxMediaType $mpxMediaTypeRepository) {
+    $this->mpxMediaTypeRepository = $mpxMediaTypeRepository;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    $mpx_media_types = $this->mpxMediaTypeRepository->findAllTypes();
+    $options = [];
+    foreach ($mpx_media_types as $mpx_media_type) {
+      $options[$mpx_media_type->id()] = $mpx_media_type->label();
+    }
+
+    $form['mpx_video_types'] = [
+      '#type' => 'checkboxes',
+      '#title' => t('mpx Video Types'),
+      '#options' => $options,
+      '#required' => TRUE,
+    ];
+    $form['queue'] = [
+      '#type' => 'submit',
+      '#value' => $this->t('Queue selected Video Types'),
+    ];
+
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function submitForm(array &$form, FormStateInterface $form_state) {
+    $types = array_filter(array_values($form_state->getValue('mpx_video_types')));
+
+    $batch_builder = new BatchBuilder();
+    $batch_builder->setTitle(t('MPX Video Imports Queueing'))
+      ->setFinishCallback([$this, 'batchFinished'])
+      ->setInitMessage(t('Starting Queueing Process.'));
```

Comment:
`Starting import`?

Reply:
Changed to "Queueing mpx videos for background imports.", just to clear any possible confusion about the import actually happening at that exact moment.